File: workhive-backend/internal/models/bid.go
package models

import "github.com/google/uuid"

type BidStatus string

const (
	BidStatusPending   BidStatus = "pending"
	BidStatusAccepted  BidStatus = "accepted"
	BidStatusRejected  BidStatus = "rejected"
	BidStatusWithdrawn BidStatus = "withdrawn"
)

type Bid struct {
	Base
	JobID        uuid.UUID `gorm:"type:uuid;not null;index:idx_bid_job_freelancer" json:"job_id"`
	FreelancerID uuid.UUID `gorm:"type:uuid;not null;index:idx_bid_job_freelancer" json:"freelancer_id"`
	Amount       float64   `gorm:"type:decimal(12,2);not null" json:"amount"`
	CoverLetter  string    `gorm:"type:text;not null" json:"cover_letter"`
	Status       BidStatus `gorm:"type:varchar(20);default:'pending';index" json:"status"`

	// associations
	Job        Job  `gorm:"foreignKey:JobID" json:"job,omitempty"`
	Freelancer User `gorm:"foreignKey:FreelancerID" json:"freelancer,omitempty"`
}

File: workhive-frontend/src/types/bid.ts
import type { UserBrief } from './auth'
import type { Job } from './job'

export interface Bid {
  id: string
  job_id: string
  freelancer_id: string
  amount: number
  cover_letter: string
  status: 'pending' | 'accepted' | 'rejected' | 'withdrawn'
  created_at: string
  updated_at: string
  freelancer?: UserBrief
  job?: Job
}

export interface SubmitBidRequest {
  amount: number
  cover_letter: string
}
